### ISSUE TITLE
"Role discovery data is unavailable" when omitting ConfigurationOptions.ClientName

### ISSUE THREAD

**Issue by urig** [StackExchange/StackExchange.Redis#499]
Based on my Stackoverflow.com question: [Azure Redis Cache: “role discovery data is unavailable”](http://stackoverflow.com/questions/39845077/azure-redis-cache-role-discovery-data-is-unavailable/39856929) and the accepted answer.

When trying to connect to an Azure Redis Cache instance from my local dev machine like so:

```
    var lazyConnection = new Lazy<ConnectionMultiplexer>(() =>
    {
        return ConnectionMultiplexer.Connect(
            $"{redisServerUrl},abortConnect=false,ssl=true,password={redisServerKey},connectTimeout=10000,syncTimeout=10000");
    });
```

I receive an `InvalidOperationException` with the message: "role discovery data is unavailable".

Adding ",name=XXX" to the connection string makes the issue go away.

Why is the `InvalidOperationException` thrown? Can a more useful error message be provided to consumers instead?

**Comment by NickCraver** [comment]
Possibly related to #498

**Comment by wangzq** [comment]
Yes, this is the same exception I got when I am debugging with all exceptions turned on. This happens when the `Microsoft.WindowsAzure.ServiceRuntime` assembly is loaded but not running in an Azure emulator or real Azure environment (such as from an Azure Service Fabric service), the fix is in my PR which checks the `IsAvailable` property first.

**Comment by NickCraver** [comment]
Fixed in #498
